Add Clarification for UDP EIH Request & Response Packet Format (#5)
Shadowsocks-NET/shadowsocks-specs: 2022-2-shadowsocks-2022-extensible-identity-headers.md

## UDP

> In UDP packets, identity headers are located between the separate header (session ID, packet ID) and AEAD ciphertext of request packets.

Reviewer: ```suggestion
In UDP request packets, identity headers are located between the separate header (session ID, packet ID) and AEAD ciphertext.

Response packets do not have identity headers.
```

```
plaintext := blake3::hash(iPSKn+1)[0..16] ^ session_id_packet_id // XOR to make it different for each packet.
identity_header := aes_encrypt(key: iPSKn, plaintext: plaintext)
```

> When iPSKs are used, the separate header MUST be encrypted with the first iPSK. Each identity processor MUST decrypt and re-encrypt the separate header with the next layer's PSK for every request packets.

Reviewer: ```suggestion
For request packets, when iPSKs are used, the separate header MUST be encrypted with the first iPSK. Each identity processor MUST decrypt and re-encrypt the separate header with the next layer's PSK.
```

> The response packets' seperate headers are encrypted with user's PSK and does not have

Reviewer: ```suggestion
For response packets, no special handling is needed for the separate header. The server encrypts the separate header with the user's PSK, per the base spec. Identity processors MUST forward response packets directly to clients without parsing or modifying them.

> identity headers.

Reviewer: ```suggestion
```
